src/community/wps-download/src/main/java/org/geoserver/wps/gs/download/ScaleToTarget.java

```diff
@@ -0,0 +1,416 @@
+/* (c) 2015 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wps.gs.download;
+
+import java.awt.geom.AffineTransform;
+import java.awt.geom.NoninvertibleTransformException;
+import java.awt.image.RenderedImage;
+import java.io.IOException;
+import java.util.List;
+
+import javax.media.jai.Interpolation;
+import javax.media.jai.InterpolationNearest;
+import javax.media.jai.JAI;
+import javax.media.jai.Warp;
+import javax.media.jai.WarpAffine;
+
+import org.geoserver.data.util.CoverageUtils;
+import org.geoserver.wcs.CoverageCleanerCallback;
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverage.grid.GridEnvelope2D;
+import org.geotools.coverage.grid.GridGeometry2D;
+import org.geotools.coverage.grid.io.AbstractGridFormat;
+import org.geotools.coverage.grid.io.GridCoverage2DReader;
+import org.geotools.coverage.grid.io.OverviewPolicy;
+import org.geotools.coverage.processing.CoverageProcessor;
+import org.geotools.factory.GeoTools;
+import org.geotools.factory.Hints;
+import org.geotools.referencing.operation.builder.GridToEnvelopeMapper;
+import org.geotools.referencing.operation.matrix.XAffineTransform;
+import org.geotools.referencing.operation.transform.ProjectiveTransform;
+import org.geotools.resources.coverage.CoverageUtilities;
+import org.jaitools.imageutils.ImageLayout2;
+import org.opengis.coverage.processing.Operation;
+import org.opengis.geometry.Envelope;
+import org.opengis.parameter.GeneralParameterDescriptor;
+import org.opengis.parameter.GeneralParameterValue;
+import org.opengis.parameter.ParameterValueGroup;
+import org.opengis.referencing.datum.PixelInCell;
+import org.opengis.referencing.operation.MathTransform;
+import org.opengis.referencing.operation.TransformException;
+
+/**
+ * Class encapsulating the logic to scale a coverage to a pre-defined target size.
+ *
+ * @author Stefano Costa, GeoSolutions
+ *
+ */
+class ScaleToTarget {
```

Comment:
It could be interesting to share this code between WPS and WCS 2.0 (since the latter is a core module WPS can lean on it) otherwise we would be duplicating things throughout the code.